cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -36,93 +36,16 @@ import (
-	if err := uploadconfig.UploadConfiguration(cfg, client); err != nil {
-		errs = append(errs, err)
-	}
-
-	// Create the new, version-branched kubelet ComponentConfig ConfigMap
-	if err := kubeletphase.CreateConfigMap(&cfg.ClusterConfiguration, client); err != nil {
-		errs = append(errs, errors.Wrap(err, "error creating kubelet configuration ConfigMap"))
-	}
-
-	// Write the new kubelet config down to disk and the env file if needed
-	if err := WriteKubeletConfigFiles(cfg, patchesDir, dryRun, out); err != nil {
-		errs = append(errs, err)
-	}
-
-	// Annotate the node with the crisocket information, sourced either from the InitConfiguration struct or
-	// --cri-socket.
-	// TODO: In the future we want to use something more official like NodeStatus or similar for detecting this properly
-	if err := patchnodephase.AnnotateCRISocket(client, cfg.NodeRegistration.Name, cfg.NodeRegistration.CRISocket); err != nil {
-		errs = append(errs, errors.Wrap(err, "error uploading crisocket"))
-	}
-
-	// Create RBAC rules that makes the bootstrap tokens able to get nodes
-	if err := nodebootstraptoken.AllowBoostrapTokensToGetNodes(client); err != nil {
-		errs = append(errs, err)
-	}
-
-	// Create/update RBAC rules that makes the bootstrap tokens able to post CSRs
-	if err := nodebootstraptoken.AllowBootstrapTokensToPostCSRs(client); err != nil {
-		errs = append(errs, err)
-	}
-
-	// Create/update RBAC rules that makes the bootstrap tokens able to get their CSRs approved automatically
-	if err := nodebootstraptoken.AutoApproveNodeBootstrapTokens(client); err != nil {
-		errs = append(errs, err)
-	}
-
-	// Create/update RBAC rules that makes the nodes to rotate certificates and get their CSRs approved automatically
-	if err := nodebootstraptoken.AutoApproveNodeCertificateRotation(client); err != nil {
-		errs = append(errs, err)
-	}
-
-	// TODO: Is this needed to do here? I think that updating cluster info should probably be separate from a normal upgrade
-	// Create the cluster-info ConfigMap with the associated RBAC rules
-	// if err := clusterinfo.CreateBootstrapConfigMapIfNotExists(client, kubeadmconstants.GetAdminKubeConfigPath()); err != nil {
-	// 	return err
-	//}
-	// Create/update RBAC rules that makes the cluster-info ConfigMap reachable
-	if err := clusterinfo.CreateClusterInfoRBACRules(client); err != nil {
-		errs = append(errs, err)
-	}
-
-	if err := PerformAddonsUpgrade(client, cfg, patchesDir, out); err != nil {
-		errs = append(errs, err)
-	}
-
-	if features.Enabled(cfg.FeatureGates, features.ControlPlaneKubeletLocalMode) {
-		if err := UpdateKubeletLocalMode(cfg, dryRun); err != nil {
-			return errors.Wrap(err, "failed to update kubelet local mode")
-		}
-	}
-
-	return errorsutil.NewAggregate(errs)
-}
-
 // PerformAddonsUpgrade performs the upgrade of the coredns and kube-proxy addons.
 func PerformAddonsUpgrade(client clientset.Interface, cfg *kubeadmapi.InitConfiguration, patchesDir string, out io.Writer) error {
```

Comment:
this function seems to include the addon upgrade which was already moved to a diff file.
maybe performaddonsupgrade can be removed.

Reply:
`PerformAddonsUpgrade` is used by `kubeadm upgrade node` control-plane phase. 
We can add a `addon` phase to `kubeadm upgrade node` in another PR.
The this func can be removed safely.